### PR TITLE
feat: LT-2032 - exhibition details

### PR DESCRIPTION
**Pull request by Mark-Spiers-kainos**

### CODE REVIEW

File: applications/views/common.py [hunk 1]
```diff
@@ -368,3 +370,26 @@ def init(self, request, **kwargs):
...
+        self.form = exhibition_details_form(self.object_pk)
+        self.action = post_exhibition
+
+    def get_data(self):
+        data = self.data
+        date_fields = ["first_exhibition_date", "required_by_date"]
+        for field in date_fields:
+            if data.get(field, False):
+                date_split = data[field].split("-")
+                data[field + "day"] = date_split[2]
```
Comment (BenAndrew-Kainos) [inline applications/views/common.py]: Could write `data[field + "year"],   data[field + "month"], data[field + "day"] = date_split`

File: core/services.py [hunk 2]
```diff
@@ -28,6 +29,25 @@ def get_units(request):
     return [Option(key, value) for key, value in data.items()]
 
 
+def get_item_types(request):
+    data = get(request, ITEM_TYPES_URL).json().get("item_types")
+    options = []
+    for k, v in data.items():
```
Comment (BenAndrew-Kainos) [inline core/services.py]: k, v is not a good variable name

File: core/services.py [hunk 3]
```diff
@@ -28,6 +29,25 @@ def get_units(request):
...
+        if k == "other":
+            options.append(
+                Option(
+                    key=k,
+                    value=v,
+                    components=[
+                        TextArea(title="", description="", name="other_item_type", extras={"max_length": 100},),
```
Comment (BenAndrew-Kainos) [inline core/services.py]: do you need title="", description="

File: ui_automation_tests/pages/standard_application/goods.py [hunk 4]
```diff
@@ -54,3 +55,7 @@ def get_good_control_codes(self):
 
     def get_goods_count(self):
         return len(self.driver.find_elements_by_css_selector(self.GOOD_ENTRY))
+
+    def click_add_to_application(self):
+        # Click the "Add to application" link on the first good
+        self.driver.find_element_by_css_selector(self.ADD_TO_APPLICATION_CSS).click()
```
Comment (BenAndrew-Kainos) [inline ui_automation_tests/pages/standard_application/goods.py]: Could we give this add to application button an id instead please

File: applications/forms/common.py [hunk 5]
```diff
@@ -89,3 +100,33 @@ def application_copy_form(application_type=None):
...
+
+
+def exhibition_details_form(application_id):
+    return Form(
+        title="Exhibition details",
+        questions=[
+            TextInput(title="Name", name="title"),
+            DateInput(
+                title="Exhibition start date",
+                description="For example, 12 11 2020",
```
[the remark this answers is not in the document]
Comment (rorysproule1) [reply applications/forms/common.py]: Strings added

File: applications/forms/goods.py [hunk 6]
```diff
@@ -9,47 +14,66 @@
...
-                        title=AddGoodToApplicationForm.Units.TITLE,
-                        description=AddGoodToApplicationForm.Units.DESCRIPTION,
-                        name="unit",
-                        options=units,
-                    ),
-                ]
+def good_on_application_form(request, good, sub_case_type, application_id):
+
+    if sub_case_type["key"] != EXHIBITION:
+        return Form(
```
[the remark this answers is not in the document]
Comment (rorysproule1) [reply applications/forms/goods.py]: Split :)

File: core/services.py (hunk 2 above)
Comment (rorysproule1) [reply core/services.py]: Renamed

File: core/services.py (hunk 3 above)
Comment (rorysproule1) [reply core/services.py]: Nope, removed them there

File: applications/services.py [hunk 7]
```diff
@@ -326,3 +326,23 @@ def get_activity(request, pk):
...
     data = post(request, APPLICATIONS_URL + str(pk) + APPLICATION_COPY_URL, json=data)
     return data.json(), data.status_code
+
+
+# Exhibition
+def post_exhibition(request, pk, data):
+    post_data = data
+
+    # Convert date
+    date_fields = ["first_exhibition_date", "required_by_date"]
```
[the remark this answers is not in the document]
Comment (rorysproule1) [reply applications/services.py]: Moved

File: applications/views/common.py (hunk 1 above)
Comment (rorysproule1) [reply applications/views/common.py]: Done

File: ui_automation_tests/pages/standard_application/goods.py (hunk 4 above)
Comment (rorysproule1) [reply ui_automation_tests/pages/standard_application/goods.py]: Done

File: applications/forms/common.py [hunk 8]
```diff
@@ -89,3 +100,33 @@ def application_copy_form(application_type=None):
     return FormGroup(
         forms=[reference_name_form(), conditional((application_type == STANDARD), have_you_been_informed()),]
     )
+
+
+def exhibition_details_form(application_id):
+    return Form(
+        title=strings.Exhibition.EXHIBITION_TITLE,
+        questions=[
+            TextInput(title="Name", name="title"),
```
Comment (Mark-Spiers-kainos) [inline applications/forms/common.py]: lite_content

File: applications/forms/goods.py [hunk 9]
```diff
@@ -9,17 +14,17 @@
     CurrencyInput,
     RadioButtons,
     Option,
+    BackLink,
 )
-from lite_forms.helpers import conditional
 
 
-def good_on_application_form(good, units):
+def exhibition_good_on_application_form(request, good, application_id):
```
Comment (Mark-Spiers-kainos) [inline applications/forms/goods.py]: this is for non_exhibition

Comment (Mark-Spiers-kainos) [reply applications/forms/goods.py]: It was inversed before, have updated them to be in the correct functions

File: applications/helpers/check_your_answers.py [hunk 10]
```diff
@@ -140,18 +133,41 @@ def _convert_hmrc_query(application, editable=False):
...
-        for good in goods
-    ]
+def _convert_goods(goods, is_exhibition=False):
+    if is_exhibition:
+        return [
+            {
+                "Description": good["good"]["description"],
+                "Part number": default_na(good["good"]["part_number"]),
+                "Controlled": friendly_boolean(good["good"]["is_good_controlled"]),
+                "CLC": default_na(good["good"]["control_code"]),
```
Comment (cstobart) [inline applications/helpers/check_your_answers.py]: try using conditional() for changing what is in a dict or a ternary which is None if false

Comment (rorysproule1) [reply applications/helpers/check_your_answers.py]: Fixed.

**Pull request merged**